mlir/test/mlir-miopen-driver/e2e/lit.cfg.py

```diff
@@ -0,0 +1,121 @@
+
+# excludes: A list of directories to exclude from the testsuite. The 'Inputs'
+# subdirectories contain auxiliary inputs for various tests in their parent
+# directories.
+config.excludes = ['Inputs', 'CMakeLists.txt', 'README.txt', 'LICENSE.txt',
+                   'lit.cfg.py', 'lit.site.cfg.py']
+
+# test_source_root: The root path where tests are located.
+#config.test_source_root = os.path.dirname(__file__)
+
+# test_exec_root: The root path where tests should be run.
+#config.test_exec_root = os.path.join(config.mlir_obj_root, 'mlir', 'test')
+
+# Tweak the PATH to include the tools dir.
+llvm_config.with_environment('PATH', config.mlir_miopen_tools_dir, append_path=True)
+llvm_config.with_environment('PATH', config.lit_tools_dir, append_path=True)
+llvm_config.with_environment('PATH', config.llvm_tools_dir, append_path=True)
+
+tool_dirs = [config.mlir_miopen_tools_dir, config.mlir_tools_dir, config.llvm_tools_dir]
+tools = ['miopen-opt']
+
+# The following tools are optional
+tools.extend([
+    ToolSubst('%PYTHON', config.python_executable, unresolved='ignore'),
+    ToolSubst('%linalg_test_lib_dir', config.linalg_test_lib_dir, unresolved='ignore'),
+    ToolSubst('%mlir_runner_utils_dir', config.mlir_runner_utils_dir, unresolved='ignore'),
+    ToolSubst('%conv_validation_wrapper_library_dir', config.conv_validation_wrapper_library_dir, unresolved='fatal'),
+])
+
+llvm_config.add_tool_substitutions(tools, tool_dirs)
+
+
+# FileCheck -enable-var-scope is enabled by default in MLIR test
+# This option avoids to accidentally reuse variable across -LABEL match,
+# it can be explicitly opted-in by prefixing the variable name with $
+config.environment['FILECHECK_OPTS'] = "-enable-var-scope --allow-unused-prefixes=false"
+
+
+# LLVM can be configured with an empty default triple
+# by passing ` -DLLVM_DEFAULT_TARGET_TRIPLE="" `.
+# This is how LLVM filters tests that require the host target
+# to be available for JIT tests.
+if config.target_triple:
+    config.available_features.add('default_triple')
+
+# Add the python path for both the source and binary tree.
+# Note that presently, the python sources come from the source tree and the
+# binaries come from the build tree. This should be unified to the build tree
+# by copying/linking sources to build.
+if config.enable_bindings_python:
+    llvm_config.with_environment('PYTHONPATH', [
+        # TODO: Don't reference the llvm_obj_root here: the invariant is that
+        # the python/ must be at the same level of the lib directory
+        # where libMLIR.so is installed. This is presently not optimal from a
+        # project separation perspective and a discussion on how to better
+        # segment MLIR libraries needs to happen. See also
+        # lib/Bindings/Python/CMakeLists.txt for where this is set up.
+        os.path.join(config.llvm_obj_root, 'python'),
+    ], append_path=True)
+
+if not config.enable_rock_e2e_test or config.no_AMD_GPU:
+  config.unsupported = True
```

Comment:
I don't think lit.cfg.py and lit.site.cfg.py.in are needed here. We can do this check in lit.local.cfg.

Reply:
Since the generated tests are in the build dir, do you mean also generate the `lit.local.cfg` file into the build dir with the tests?

Reply:
I was thinking having these files in the parent folder and the local.cfg in e2e, but on the second thought, I agree with Krzysztof that we can just have e2e directly under build. There is no need to have a mlir-miopen-driver directory there.